staging/src/k8s.io/apiserver/pkg/server/config.go

```diff
@@ -188,6 +189,10 @@ type Config struct {
 	// kube-proxy, services, etc.) can reach the GenericAPIServer.
 	// If nil or 0.0.0.0, the host's default interface will be used.
 	PublicAddress net.IP
+
+	// ResourceMapper provides information about equivalent resources,
+	// and the kind associated with a given resource
+	ResourceMapper runtime.ResourceMapper
```

Comment:
also here it is unclear whether this provides or receives that information. Having two interfaces for read and write would make that much clearer.

ResourceMappingRecorder vs. ResourceMapper